pkg/build/builder/dockerutil.go

```diff
@@ -39,14 +44,18 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		opts.OutputStream = os.Stderr
 	}
 	var err error
+
 	for retries := 0; retries <= DefaultPushRetryCount; retries++ {
-		err = client.PushImage(opts, authConfig)
+		err := client.PushImage(opts, authConfig)
```

Comment:
It compiles because the `for` loop introduces a scope. `err` was declared and initialized in that scope and referenced (satisfying the compiler). The earlier `err` declared outside the `for` was not initialized, but referenced in the last `return` statement (always returning `nil`).